### PR TITLE
[WIP] End to end training of masker and painter

### PR DESCRIPTION
**Pull request by 51N84D**
If both "m" and "p" are specified,  omnigan goes into end-to-end training mode. This just means that we use the global discriminator (which is trained to distinguish between flooded and non-flooded images) to compute a loss on the painter and masker in the real domain. Let `x` be an image in the real domain.

z_x = G.encode(x)
m = masker(z_x)
z ~ N(0,1)
x_f = painter(z, m)

Then we compute the GAN loss on `x_f`

### CODE REVIEW

**Comment by 51N84D** [comment]
The code currently isn't set up to load checkpoints for the masker and painter separately though. I can work on that next

**Comment by vict0rsch** [comment]
> The code currently isn't set up to load checkpoints for the masker and painter separately though. I can work on that next

sounds good ; do you need help? with hydra integrations things should become more flexible

File: shared/trainer/defaults.yaml [hunk 1]
```diff
@@ -93,7 +94,7 @@ gen:
   m: # specific params for the mask-generation decoder
     <<: *default-gen
     output_dim: 1
-    use_advent: True
+    use_advent: False
```
Comment (vict0rsch) [inline shared/trainer/defaults.yaml:97]: is there a reason for this change of default?

Comment (51N84D) [reply shared/trainer/defaults.yaml:97]: Without advent seems to work better than with advent at the moment. The defaults should reflect our best performance right?

**Pull request closed without merging**